### ISSUE TITLE
'S3Uploader' object has no attribute 'get'

### ISSUE THREAD

**Issue by odubyk** [aws/aws-sam-cli#2501]
### Description:
After release [1.13.2](https://github.com/aws/aws-sam-cli/releases/tag/v1.13.2) the fixes https://github.com/aws/aws-sam-cli/pull/2440, `sam package` command can't upload nested template that contain `Fn::Transform`

### Steps to reproduce:
define `Fn::Transform` in a template that uses for nested stack (AWS::CloudFormation::Stack) and run `sam package`

### Observed result:
```
Error: Unable to upload artifact service-app.yml referenced by TemplateURL parameter of ApplicationService resource.
'S3Uploader' object has no attribute 'get'
```

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: ununtu 16.04
2. `sam --version`: >=1.13.2

`Add --debug flag to command you are running`

**Comment by jfuss** [comment]
@odubyk Thought that PR fixes the issue. Can you provide a repro template example for us?

**Comment by odubyk** [comment]
$ cat template.yml 
```
AWSTemplateFormatVersion: "2010-09-09"

Resources:
  ApplicationService:
    Type: AWS::CloudFormation::Stack
    Properties:
      TemplateURL: service-app.yml
      Parameters:
        Secret: test
```

$ cat service-app.yml 
```
AWSTemplateFormatVersion: "2010-09-09"
Parameters:
  Secret:
    Type: String

Resources:
  TaskDefinition:
    Type: AWS::ECS::TaskDefinition
    Properties:
      Family: test
      NetworkMode: awsvpc
      RequiresCompatibilities:
        - FARGATE
      ContainerDefinitions:
        - Name: test
          LogConfiguration:
          'Fn::Transform':
            Name: AWS::Include
            Parameters:
              Location: env-secrets.yml
          Command: [ 'ps' ]
```

$ cat env-secrets.yml 
```
Environment:
  - name: RAILS_ROOT
    Value: /app
secrets:
  - name: RAILS_ENV
    valueFrom:
      Fn::Sub: '${Secret}:RAILS_ENV::'
```

command:
```
sam --debug package --s3-bucket sam-deployment-bucket --template-file template.yml --output-template-file packaged.yml
```

aws-sam-cli:
- v1.13.1 - works
- v1.13.2 and next - doesn't work

**Comment by aahung** [comment]
This should be fixed by https://github.com/aws/aws-sam-cli/pull/2532 which will be included in our next release.

`Template`'s `uploader` object could either be a `dict` or `S3Uploader`, and the error "'S3Uploader' object has no attribute 'get'" is raised when `uploader` is `S3Uploader`. This usually happens in nested stacks.

**Comment by BatmanAoD** [comment]
This appears to also affect the `AWS::Serverless::API` resource type.

**Comment by aahung** [comment]
v1.16.0 is released, please reopen if the issue still persists

**Comment by aczire** [comment]
Still waiting for this release to hit codebuild images. Any idea when it will be available in "aws/codebuild/standard:4.0" image?
